core/src/main/scala/better/files/UnicodeCharset.scala

```diff
@@ -51,7 +52,8 @@ class UnicodeDecoder(defaultCharset: Charset) extends CharsetDecoder(defaultChar
     }
   }
 
-  override def decodeLoop(in: ByteBuffer, out: CharBuffer) = decode(in = in, out = out, candidates = bomTable.keySet)
+  override def decodeLoop(in: ByteBuffer | Null, out: CharBuffer | Null) =
+    decode(in = in.nn, out = out.nn, candidates = bomTable.keySet)
```

Comment:
**Overridden java arguments are never null**
Argument should never be null, immediately remove nullness of parameter.